### PR TITLE
Ported kinesis exporter from Omnition

### PR DESCRIPTION
**Pull request by owais**
Porting the existing kinesis exporter from Omnition's Otel distribution
to contrib.

Porting from: https://github.com/Omnition/omnition-opentelemetry-collector/tree/master/exporter/kinesis

Currently exports to Kinesis in Jaeger proto format only. Support for other formats can be added later.

### CODE REVIEW

**Comment by owais** [comment]
Looking at CI failures.

File: go.mod [hunk 1]
```diff
@@ -22,3 +24,5 @@ require (
 replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuremonitorexporter v0.0.0 => ./exporter/azuremonitorexporter
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sapmexporter => ./exporter/sapmexporter
```
Comment (tigrannajaryan) [inline go.mod]: Can you remove these "replace" directives that are no longer needed?

**Review by tigrannajaryan — APPROVED** [review]
I had reviewed this code earlier in its original repo, approving.

File: go.mod (hunk 1 above)
Comment (owais) [reply go.mod]: I can but first we need to merge this or we'll need another PR I think. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/59

File: exporter/kinesisexporter/config_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+			FlushIntervalSeconds: 3,
+			MaxBytesPerBatch:     4,
+			MaxBytesPerSpan:      5,
+		},
+	)
+}
+
+func TestConfigCheck(t *testing.T) {
+	cfg := (&Factory{}).CreateDefaultConfig()
+	assert.NoError(t, configcheck.ValidateConfig(cfg))
```
Comment (pjanotti) [inline exporter/kinesisexporter/config_test.go]: 👍

File: exporter/kinesisexporter/config.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	Region          string `mapstructure:"region"`
+	Role            string `mapstructure:"role"`
+}
+
+// KPLConfig contains kinesis producer library related config to controls things
+// like aggregation, batching, connections, retries, etc.
+type KPLConfig struct {
+	AggregateBatchCount  int `mapstructure:"aggregate_batch_count"`
+	AggregateBatchSize   int `mapstructure:"aggregate_batch_size"`
+	BatchSize            int `mapstructure:"batch_size, omitempty"`
```
Comment (pjanotti) [inline exporter/kinesisexporter/config.go]: nit: remove `omitempty`

File: go.mod (hunk 1 above)
Comment (pjanotti) [reply go.mod]: I kind of prefer to go in two PRs but up to you.

**Review by pjanotti — APPROVED** [review]
It will be nice to comment on the configuration fields but due to the number of them let's do it separately. I also had looked at this code before but tried to look again... single nit, LGTM.

**Comment by owais** [comment]
Updated.

**Pull request merged**